docs/migration-guide/to-16.md

```diff
@@ -2,93 +2,208 @@
+  utils: { report, validateOptions },
+} = stylelint;
 
-See also our [developer guide](../developer-guide/syntaxes.md) with examples of ESM syntax.
+const ruleFunction = (primary, secondaryOptions) => { /* .. */ };
 
-### Added support for ESM custom formatters
+ruleFunction.ruleName = ruleName;
+ruleFunction.messages = messages;
+ruleFunction.meta = meta;
 
-You can now specify an ESM formatter via the [`--custom-formatter`](../user-guide/options.md#formatter) CLI flag. For example:
+- module.exports = createPlugin(ruleName, ruleFunction);
++ export default createPlugin(ruleName, ruleFunction);
+```
 
-```js
-// your-own-formatter.js
-export default function yourOwnFormatter(results) {
-  // return a string from results...
+We've published a new version of [Jest preset](https://github.com/stylelint/jest-preset-stylelint) for plugin tests. If you use it, you should update your npm-run-script:
+
+```diff json
+{
+  "scripts": {
+-    "test": "jest",
++    "test": "cross-env NODE_OPTIONS=\"--experimental-vm-modules --no-warnings\" jest --runInBand",
+  }
 }
 ```
 
-```shell
-stylelint --custom-formatter=./your-own-formatter.js "*.css"
+#### `stylelint.lint()`
+
+For example, to migrate your code that uses `stylelint.lint()` to use `import` and top-level `await`:
+
+```diff js
+- const stylelint = require("stylelint");
++ import stylelint from "stylelint";
+
+- stylelint.lint(options).then((result) => { console.log(result); });
++ const result = await stylelint.lint(options);
++ console.log(result);
+```
+
+You'll find more details about the [ESM Node.js API in the user guide](../user-guide/node-api.md).
+
+Using the CommonJS Node.js API will trigger a deprecation warning. If you're not quite ready to migrate to ESM yet, you can use the [`quietDeprecationWarnings`](../user-guide/options.md#quietdeprecationwarnings) option to hide the warning.
+
+```diff js
+const stylelint = require("stylelint");
+
+const resultPromise = stylelint.lint({
+  /* .. */
++ quietDeprecationWarnings: true
+});
+```
+
+### Refactored to use `.mjs` and `.cjs` extensions internally
+
+We now use `.mjs` and `.cjs` extensions internally to support a hybrid package. This change doesn't affect our public API but will affect you if your plugin `require`s internal files.
+
+We recommended copying files to your project instead of importing them as we'll remove access to them in the next major release.
+
+However, you can unsafely continue to `import` or `require` the files before the next major release by updating your imports:
+
+```diff js
+// ESM
+-import('stylelint/lib/utils/typeGuards.js');
++import('stylelint/lib/utils/typeGuards.mjs');
+
+// CommonJS
+-require('stylelint/lib/utils/typeGuards.js');
++require('stylelint/lib/utils/typeGuards.cjs');
 ```
 
 ## Breaking changes
 
-This release contains breaking changes. We've:
+We've:
 
 - removed deprecated stylistic rules
 - removed support for Node.js less than 18.12.0
-- changed `.js` extension to `.mjs` and `.cjs`
```

Comment:
question

If it's not a breaking change, do we downgrade https://github.com/stylelint/stylelint/blob/main/.changeset/loud-insects-camp.md to a minor?
i.e. so that the CHANGELOG won't have _(BREAKING)_ next to it